utils.py

```diff
@@ -0,0 +1,54 @@
+import os
+import re
+import hashlib
+
+def e(s):
+    if type(s) == bytes:
+        return s
+    return s.encode('utf-8')
+
+def mkid(s):
+    return hashlib.sha1(e(s)).hexdigest()[:2*4]
+
+def running_in_tools_labs():
+    return (os.path.exists('/etc/wmflabs-project') or
+        os.path.exists('/etc/wmcs-project'))
+
+def text_to_word_list(text):
```

Comment:
@mirrys Could this be replaced with https://www.nltk.org/api/nltk.tokenize.html#nltk.tokenize.punkt.PunktLanguageVars.word_tokenize instead?

Reply:
I wouldn't advise it unless the function proposed is performing the same word processing as the one in the original script.

Reply:
Yeah you're right, actually it would be a bad idea for us to change it here now. @AikoChou let's just leave this as-is for now please :)

For the future, it's not great to have to maintain a list of "cleanup regexes" (per language?!) in multiple places (in the scripts that train the model _and_ any clients that want to use it, such as Citation Detective). We should be thinking about stuff like:

- Moving this function to a library that is used for interacting with the model. This library is used for training new versions of the model and accessible to clients (such as Citation Detective) that want to use it, without having to duplicate code.
- Replacing these cleanup functions with tokenization/stemming/lemming functions from established Python packages where possible, so we don't have to maintain our own versions.

...though all of that is probably out of the scope of this Outreachy project.

Reply:
> I wouldn't advise it unless the function proposed is performing the same word processing as the one in the original script.

The 'word_tokenize' provided by nltk library is different from 'text_to_word_list' when processing contracted words like `can't` `would've` `don't` `they'd`. 'word_tokenize' does not convert them to original form. Also, when it comes to `12.1/km²`, 'word_tokenize' tends to preserve the whole but 'text_to_word_list' divides it as `12` `1` and `km`. 

I agree we should leave it not change for now because the model is trained based on the results from this function. But I feel this function may need to be improved as I found it does not remove punctuations like `;` `:` `-` and split words like `U.N.` and decimal numbers.

> For the future, it's not great to have to maintain a list of "cleanup regexes" (per language?!) in multiple places (in the scripts that train the model _and_ any clients that want to use it, such as Citation Detective). We should be thinking about stuff like:
> * Moving this function to a library that is used for interacting with the model. This library is used for training new versions of the model and accessible to clients (such as Citation Detective) that want to use it, without having to duplicate code.
> * Replacing these cleanup functions with tokenization/stemming/lemming functions from established Python packages where possible, so we don't have to maintain our own versions.

I will file an issue for this as future work in case we forget :)